Plash/Utilities.swift

```diff
@@ -1502,6 +1506,50 @@ extension WKWebView {
 			completion($0 == .OK ? openPanel.urls : nil)
 		}
 	}
+
+	// Can be tested at https://jigsaw.w3.org/HTTP/Basic/ with `guest` as username and password.
+	/// Default handler for websites requiring basic authentication. To be used in `WKDelegate`.
+	func defaultAuthChallengeHandler(challenge: URLAuthenticationChallenge, completion: @escaping (URLSession.AuthChallengeDisposition, URLCredential?) -> Void) {
+		guard
+			let host = url?.host,
+			challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodHTTPBasic
+				|| challenge.protectionSpace.authenticationMethod == NSURLAuthenticationMethodHTTPDigest
+		else {
+			completion(.performDefaultHandling, nil)
+			return
+		}
+
+		let alert = NSAlert()
+		alert.messageText = "Log in to \(host)"
+		alert.addButton(withTitle: "Log In")
+		alert.addButton(withTitle: "Cancel")
+
+		let view = NSView(frame: CGRect(x: 0, y: 0, width: 200, height: 49))
+		alert.accessoryView = view
+
+		// TODO: Add `.contentType` to these when on Xcode 12.
+		let username = AutofocusedTextField(frame: CGRect(x: 0, y: 27, width: 200, height: 21))
+		username.placeholderString = "Username"
```

Comment:
Is this enough for VoiceOver to know what the text field's name is?

Reply:
That's a good observation. Yes, I tested this already. It works fine with Voice Over.